packages/bootstrap-4/src/FieldTemplate/WrapIfAdditional.tsx

```diff
@@ -0,0 +1,80 @@
+
+import React from "react";
+
+import { utils } from "@rjsf/core";
+import { JSONSchema7 } from "json-schema";
+
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Form from "react-bootstrap/Form";
+
+import IconButton from "../IconButton/IconButton";
+
+const { ADDITIONAL_PROPERTY_FLAG } = utils;
+
+type WrapIfAdditionalProps = {
+  children: React.ReactElement;
+  classNames: string;
+  disabled: boolean;
+  id: string;
+  label: string;
+  onDropPropertyClick: (index: string) => (event?: any) => void;
+  onKeyChange: (index: string) => (event?: any) => void;
+  readonly: boolean;
+  required: boolean;
+  schema: JSONSchema7;
+};
+
+const WrapIfAdditional = ({
+  children,
+  disabled,
+  id,
+  label,
+  onDropPropertyClick,
+  onKeyChange,
+  readonly,
+  required,
+  schema,
+}: WrapIfAdditionalProps) => {
+  const keyLabel = `${label} Key`; // i18n ?
+  const additional = schema.hasOwnProperty(ADDITIONAL_PROPERTY_FLAG);
+
+  if (!additional) {
+    return children;
+  }
+
+  const handleBlur = ({ target }: React.FocusEvent<HTMLInputElement>) =>
+    onKeyChange(target.value);
+
+  return (
+    <Row key={`${id}-key`}>
+      <Col xs={5}>
+        <Form.Group>
+          <Form.Label>{keyLabel}</Form.Label>
+          <Form.Control
+            required={required}
+            defaultValue={label}
+            disabled={disabled || readonly}
+            id={`${id}-key`}
+            name={`${id}-key`}
+            onBlur={!readonly ? handleBlur : undefined}
+            type="text"
+          />
+        </Form.Group>
+      </Col>
+      <Col xs={6}>
+        {children}
+      </Col>
+      <Col xs={1} className="py-4">
+        <IconButton
+          icon="remove"
```

Comment:
potentially see if we could make it wider / if bootstrap has an easy class name to do so, then let's do it.

Reply:
Added prop `block={true}`